docker/pulsar-all/pom.xml

```diff
@@ -161,5 +169,29 @@
         </plugins>
       </build>
     </profile>
+
+    <profile>
+      <id>docker-push</id>
```

Comment:
@merlimat Shall this change affect existing docker push scripts and the release process?

Reply:
I have the same concern, we only use https://github.com/apache/pulsar/blob/ed9cca6d5dd947c741b6d6fc579b7d526ba0274c/docker/publish.sh to publish the docker image, so I think this profile is unnecessary.

Reply:
The point here is that we need a profile that does build the multi-arch when doing a release. 

It's also important that the build & push are done in one step so that buildx can push a single tag with 2 separate images.

Yes, the release process would need to be adjust a little bit.

Reply:
buildx can merge two images under one tag in a separate step though - https://github.com/korandoru/hawkeye/blob/main/.github/actions/docker-release/action.yml

But it's fine to give the current way a try :)